drake/systems/DrakeSystem.cpp

```diff
@@ -0,0 +1,138 @@
+
+#include <algorithm>
+#include "DrakeSystem.h"
+
+using namespace std;
+using namespace Eigen;
+
+DrakeSystem::DrakeSystem(const std::string &_name, const std::shared_ptr<CoordinateFrame>& _continuous_state_frame,
+                         const std::shared_ptr<CoordinateFrame>& _discrete_state_frame,
+                         const std::shared_ptr<CoordinateFrame>& _input_frame,
+                         const std::shared_ptr<CoordinateFrame>& _output_frame)
+  : name(_name),
+    continuous_state_frame(_continuous_state_frame),
+    discrete_state_frame(_discrete_state_frame),
+    input_frame(_input_frame),
+    output_frame(_output_frame) {
+  if (!continuous_state_frame) continuous_state_frame = make_shared<CoordinateFrame>(name+"ContinuousState");
+  if (!discrete_state_frame) discrete_state_frame = make_shared<CoordinateFrame>(name+"DiscreteState");
+  if (!input_frame) input_frame = make_shared<CoordinateFrame>(name+"Input");
+  if (!output_frame) output_frame = make_shared<CoordinateFrame>(name+"Output");
+  state_frame = make_shared<MultiCoordinateFrame>(name+"State",initializer_list<CoordinateFramePtr>({continuous_state_frame,discrete_state_frame}));
+}
+
+DrakeSystem::DrakeSystem(const std::string &_name, unsigned int num_continuous_states, unsigned int num_discrete_states,
+                         unsigned int num_inputs, unsigned int num_outputs)
+        : name(_name),
+          input_frame(nullptr),
+          continuous_state_frame(nullptr),
+          discrete_state_frame(nullptr),
+          output_frame(nullptr) {
```

Comment:
i'm not used to putting what felt like code up there.  but i guess you're right.  i've made the change.